packages/edit-site/src/components/add-new-template/utils.js

```diff
@@ -80,10 +80,18 @@ const usePublicPostTypes = () => {
 	);
 	return useMemo( () => {
 		const excludedPostTypes = [ 'attachment' ];
-		return postTypes?.filter(
-			( { viewable, slug } ) =>
-				viewable && ! excludedPostTypes.includes( slug )
-		);
+		return postTypes
+			?.filter(
+				( { viewable, slug } ) =>
+					viewable && ! excludedPostTypes.includes( slug )
+			)
+			.sort( ( a, b ) => {
+				if ( a.slug === 'post' || b.slug === 'post' ) {
+					return 0;
+				}
```

Comment:
```suggestion
				// Sort post types alphabetically by name,
				// but exclude the built-in 'post' type from sorting.
				if ( a.slug === 'post' || b.slug === 'post' ) {
					return 0;
				}
```

Nit: This code is hard to understand at first glance and may need some comments. What do you think?